contrib/packs/actions/pack_mgmt/download.py

```diff
@@ -145,18 +150,26 @@ def _move_packs(self, abs_repo_base, packs, abs_local_path, subtree):
                 self.logger.debug('Moving pack from %s to %s.', abs_pack_temp_location, to)
                 shutil.move(abs_pack_temp_location, to)
                 # post move fix all permissions.
-                DownloadGitRepoAction._apply_pack_permissions(to)
+                try:
+                    self._apply_pack_permissions(pack_path=dest_pack_path)
+                except:
+                    self.loggger.exception('Damn It!')
                 message = 'Success.'
             elif message:
                 message = 'Failure : %s' % message
             result[pack] = (desired, message)
         return result
 
-    @staticmethod
-    def _apply_pack_permissions(pack_path):
+    def _apply_pack_permissions(self, pack_path):
         """
         Will recursively apply permission 770 to pack and its contents.
         """
+        # 1. switch owner group to configuered group
+        pack_group = self.config.get(PACK_GROUP_CFG_KEY, None)
+        if pack_group:
+            subprocess.call(['sudo', 'chgrp', '-R', pack_group, pack_path])
```

Comment:
If it will, we should leave the config empty.

And even if it's defined, we should probably throw and fail fast if an invalid / in-existent group is referenced.